### PR TITLE
Replace `github.com/pborman/uuid` with `github.com/google/uuid`

### PR DESCRIPTION
**Pull request by Juneezee**
**What type of PR is this:**

/area refactoring

**What does this PR do / why we need it:**

`github.com/pborman/uuid` is no longer being actively maintained (last release was 2019). So we should just use the newer `github.com/google/uuid`.

image

**Which issue(s) this PR fixes:**

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/segment/segment.go [hunk 1]
```diff
@@ -215,11 +216,12 @@ func GetUserIdentity(telemetryFilePath string) (string, error) {
 	}
 
 	// check if the id is a valid uuid, if not, nil is returned
-	if uuid.Parse(strings.TrimSpace(string(id))) == nil {
-		id = []byte(uuid.NewRandom().String())
-		if err := os.WriteFile(telemetryFilePath, id, 0600); err != nil {
+	if _, err := uuid.ParseBytes(bytes.TrimSpace(id)); err != nil {
+		uid := uuid.New()
```
Comment (rm3l) [inline pkg/segment/segment.go:220]: `uuid.New()` would panic if it couldn't return an UUID. Could you please use `uuid.NewRandom()` instead and handle the error instead?

File: pkg/segment/segment.go [hunk 2]
```diff
@@ -215,11 +216,12 @@ func GetUserIdentity(telemetryFilePath string) (string, error) {
 	}
 
 	// check if the id is a valid uuid, if not, nil is returned
```
Comment (rm3l) [inline pkg/segment/segment.go:218]: Also, please update this comment above, which is no longer accurate.

File: pkg/segment/segment.go [hunk 3]
```diff
@@ -215,11 +216,12 @@ func GetUserIdentity(telemetryFilePath string) (string, error) {
...
-	if uuid.Parse(strings.TrimSpace(string(id))) == nil {
-		id = []byte(uuid.NewRandom().String())
-		if err := os.WriteFile(telemetryFilePath, id, 0600); err != nil {
+	if _, err := uuid.ParseBytes(bytes.TrimSpace(id)); err != nil {
+		uid := uuid.New()
+		if err := os.WriteFile(telemetryFilePath, uid[:], 0o600); err != nil {
```
Comment (rm3l) [inline pkg/segment/segment.go:221]: I noticed that the content of the file generated does not contain the string form of the UUID, e.g.:
```
$ file /tmp/telemetry823456139
/tmp/telemetry823456139: data

$ cat /tmp/telemetry823456139
�5�!��M)�-V�9`^V^^
```

Could you convert the returned UUID into a string before persisting it into the file?
```diff
diff --git a/pkg/segment/segment.go b/pkg/segment/segment.go
index f1b30276d..fa6d3dd46 100644
--- a/pkg/segment/segment.go
+++ b/pkg/segment/segment.go
@@ -217,11 +217,14 @@ func GetUserIdentity(telemetryFilePath string) (string, error) {
 
        // check if the id is a valid uuid, if not, nil is returned
        if _, err := uuid.ParseBytes(bytes.TrimSpace(id)); err != nil {
-               uid := uuid.New()
-               if err := os.WriteFile(telemetryFilePath, uid[:], 0o600); err != nil {
+               u, uErr := uuid.NewRandom()
+               if uErr != nil {
+                       return "", fmt.Errorf("failed to generate anonymous ID for telemetry: %w", uErr)
+               }
+               id = []byte(u.String())
+               if err := os.WriteFile(telemetryFilePath, id, 0600); err != nil {
                        return "", err
                }
-               id = uid[:]
        }
        return strings.TrimSpace(string(id)), nil
 }
```

**Review by rm3l — CHANGES_REQUESTED** [review]

File: pkg/segment/segment.go (hunk 1 above)
Comment (Juneezee) [reply pkg/segment/segment.go]: Fixed in https://github.com/redhat-developer/odo/pull/7173/commits/e4bb039e7a7d89b58f9d12865ec164cf49d571cf.

File: pkg/segment/segment.go (hunk 2 above)
Comment (Juneezee) [reply pkg/segment/segment.go]: Fixed in https://github.com/redhat-developer/odo/pull/7173/commits/e4bb039e7a7d89b58f9d12865ec164cf49d571cf.

File: pkg/segment/segment.go (hunk 3 above)
Comment (Juneezee) [reply pkg/segment/segment.go]: Thanks! Fixed in https://github.com/redhat-developer/odo/pull/7173/commits/e4bb039e7a7d89b58f9d12865ec164cf49d571cf.

**Pull request merged**